### DISCUSSION TITLE
运行锁定设置同一时间最多只能运行一个构建任务优化

### DISCUSSION THREAD

**Discussion by yanwen249** [Tencent/bk-ci#6448] [Ideas]
问题描述：运行锁定设置同一时间最多只能运行一个构建任务，当执行任务数量超过最大排队数量时会执行成功，但同时会自动取消掉任务队列中队首任务，而且无任何提示。
![image](https://user-images.githubusercontent.com/1404391/160574783-b222071d-580b-4eca-9d78-7b0601873434.png)

产生的问题：多人协助场景下，可能出现排队中的任务被其他人顶掉，而且无任何感知和提示，将造成时间浪费。本意使用这个功能的目的是为了在当前有任务运行时，可以排队进行，自己可以切换去做其他事情等待任务自动完成，但是目前的方案不符合常理，不仅达不到提效的目的还将浪费更多的时间

优化方向：超过最大排队数量时，提交任务提示队列已满，不进入排队队列

**Comment by iccengan** [comment]
CI一般都会更关注同个分支的最新提交版本，所以默认是会踢掉旧任务并将新任务置入队列。
@yanwen249 这边是什么场景下需要保留旧的任务并踢掉新的任务呢？
